plugins/proxy-backend/src/service/router.ts

```diff
@@ -191,6 +191,37 @@ export function buildMiddleware(
   return createProxyMiddleware(filter, fullConfig);
 }
 
+function readProxyConfig(config: Config, logger: Logger): unknown {
+  const endpoints = config.getOptional('proxy.endpoints');
```

Comment:
could this have been `getOptionalConfig(...)?.get()` to have all of its internal validation mechanics do the heavy type lifting?